lib/ecto/query/builder/limit.ex

```diff
@@ -0,0 +1,58 @@
+import Kernel, except: [apply: 2]
+
+defmodule Ecto.Query.Builder.Limit do
+  @moduledoc false
+
+  alias Ecto.Query.Builder
+
+  @doc """
+  Builds a quoted expression.
+
+  The quoted expression should evaluate to a query at runtime.
+  If possible, it does all calculations at compile time to avoid
+  runtime work.
+  """
+  @spec build(Macro.t, [Macro.t], Macro.t, boolean(), Macro.Env.t) :: Macro.t
+  def build(query, binding, expr, with_ties, env) when is_boolean(with_ties) do
+    {query, binding} = Builder.escape_binding(query, binding, env)
+    {expr, {params, _acc}} = Builder.escape(expr, :integer, {[], %{}}, binding, env)
+    params = Builder.escape_params(params)
+
+    if contains_variable?(expr) do
+      Builder.error! "query variables are not allowed in limit expressions"
+    end
+
+    limit = quote do: %Ecto.Query.LimitExpr{
+                        expr: unquote(expr),
+                        with_ties: unquote(with_ties),
```

Comment:
Based on the previous discussion, this would be something like:

```suggestion
                        with_ties: Keyword.get(unquote(opts), :with_ties, false),
```

But if we go with the separate function, then I think we don't need to change this at all. :)